### PR TITLE
Support Modular Auto Generation

### PR DESCRIPTION
**Pull request by wanlwanl**
Fix: https://github.com/Azure/azure-sdk-tools/issues/8307

### CODE REVIEW

File: tools/js-sdk-release-tools/src/utils/generateInputUtils.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+    const resolvedRelativeTspFolder = Array.isArray(typespecProjectFolder) ? typespecProjectFolder[0] : typespecProjectFolder as string;
+    const tspFolderFromSpecRoot = path.join(specFolder, resolvedRelativeTspFolder);
+    const tspConfigPath = path.join(tspFolderFromSpecRoot, 'tspconfig.yaml');
+    if (!(await exists(tspConfigPath))) {
+        return false;
+    }
+
+    const tspConfig = await loadTspConfig(tspFolderFromSpecRoot);
+    if (tspConfig?.options?.['@azure-tools/typespec-ts']?.['isModularLibrary'] !== true) {
```
Comment (qiaozha) [inline tools/js-sdk-release-tools/src/utils/generateInputUtils.ts:24]: how about the case where data plane has enabled isModularLibrary?

Comment (qiaozha) [reply tools/js-sdk-release-tools/src/utils/generateInputUtils.ts:24]: some of the configurations may not be appilicable for data plane.

File: tools/js-sdk-release-tools/src/test/pipeline/sdkGeneration.test.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
+import path from 'path';
+import { describe, expect, test } from 'vitest';
+import { getRandomInt } from '../utils/utils';
+import { ensureDir, remove, writeFile } from 'fs-extra';
+import { stringify } from 'yaml';
+import { SDKType } from '../../common/types';
+import { parseInputJson } from '../../utils/generateInputUtils';
+
+describe('Auto SDK generation path test', () => {
+    test('high level client generation', async () => {
```
Comment (qiaozha) [inline tools/js-sdk-release-tools/src/test/pipeline/sdkGeneration.test.ts:10]: I wonder if we could change the indent as two spaces? maybe not in this PR.

Comment (wanlwanl) [reply tools/js-sdk-release-tools/src/test/pipeline/sdkGeneration.test.ts:10]: Sure, I'll format the whole js-sdk-release-tool project in another pr.

File: tools/js-sdk-release-tools/src/utils/generateInputUtils.ts (hunk 1 above)
Comment (wanlwanl) [reply tools/js-sdk-release-tools/src/utils/generateInputUtils.ts:24]: synced offline. Support management plane for now. Created a [issue ](https://github.com/Azure/azure-sdk-tools/issues/8935) to track

**Review by qiaozha — APPROVED** [review]

**Pull request merged**